### PR TITLE
Fixes #1506 OpenLayers and Leaflet vector different default styles 

### LINKED ISSUE #1506: OpenLayers and Leaflet vector different default styles
Leaflet and Openlayers have different default styles. 
**Polygon:**
leaflet
image 
OpenLayers
image
**Line:**
Leaflet
image

image

### PR DESCRIPTION
**Pull request by allyoucanmap**
## Description
This PR introduces the posiibilities to customize styles for geometries of searched feature (geocoder).
Added a default style to align OL with Leaflet.

custom configuration:
```
{
    "name": "Search",
    "cfg": {
        "resultsStyle": {
            "iconUrl": "https://cdn.rawgit.com/pointhi/leaflet-color-markers/master/img/marker-icon-red.png",
            "shadowUrl": "https://cdnjs.cloudflare.com/ajax/libs/leaflet/0.7.7/images/marker-shadow.png",
            "iconSize": [
                25,
                41
            ],
            "iconAnchor": [
                12,
                41
            ],
            "popupAnchor": [
                1,
                -34
            ],
            "shadowSize": [
                41,
                41
            ],
            "color": "#ff0000",
            "weight": 4,
            "dashArray": "",
            "fillColor": "rgba(255, 0, 255, 0.2)",
            "fillOpacity": 1.0,
            "LineString": {
                // custom style for LineString, it overrides deafult/general style (optional)
            },
            "MultiLineString": {
                // custom style for MultiLineString, it overrides deafult/general style (optional)
            },
            "Polygon": {
                // custom style for Polygon, it overrides deafult/general style (optional)
            },
            "MultiPolygon": {
                // custom style for MultiPolygon, it overrides deafult/general style (optional)
            }
        }
    }
}
```

## Issues
 - Fix #1506

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)

 - [x] Feature

**What is the current behavior?** (You can also link to an open issue here)
It's not possible to customize the style of resulted geometries from search (geocoder).
Deafult styles of OL and Leaflet are different.

**What is the new behavior?**
It's possible to customize the style of resulted geometries from search (geocoder).
Added a default style to align OL with Leaflet.

**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: web/client/components/map/leaflet/Feature.jsx [hunk 1]
```diff
@@ -71,13 +71,13 @@ const geometryToLayer = function(geojson, options) {
 
     case 'LineString':
         latlngs = coordsToLatLngs(coords, geometry.type === 'LineString' ? 0 : 1, coordsToLatLng);
-        layer = new L.Polyline(latlngs, options.style);
+        layer = new L.Polyline(latlngs, options.style && options.style.LineString || options.style);
```
Comment (mbarto) [inline web/client/components/map/leaflet/Feature.jsx]: We should use the type to select the style (options.style && options.style[type] || options.style), and possibly move it outside the switch:

```
const style = options.style && options.style[type] || options.style;
....
layer = new ....(..., style);
```

**Comment by mbarto** [comment]
@allyoucanmap looks good, just a minor comment

File: web/client/plugins/Search.jsx [hunk 2]
```diff
@@ -144,6 +171,13 @@ class extends React.Component {
         searchOptions: {
             services: [{type: "nominatim"}]
         },
+        resultsStyle: {
+            color: '#3388ff',
+            weight: 4,
+            dashArray: '',
+            fillColor: 'rgba(51, 136, 255, 0.2)',
+            fillOpacity: 1.0
```
Comment (MV88) [inline web/client/plugins/Search.jsx]: fill opacity should be 0.2 or fill color needs to be changed

File: web/client/selectors/layers.js [hunk 3]
```diff
@@ -22,6 +22,15 @@ const geoColderSelector = state => state.search && state.search;
 // TODO currently loading flag causes a re-creation of the selector on any pan
 // to avoid this separate loading from the layer object
 
+const deafaultIconStyle = {
+    iconUrl: "https://cdn.rawgit.com/pointhi/leaflet-color-markers/master/img/marker-icon-red.png",
+    shadowUrl: 'https://cdnjs.cloudflare.com/ajax/libs/leaflet/0.7.7/images/marker-shadow.png',
```
Comment (MV88) [inline web/client/selectors/layers.js]: why 0.7.7 ?

File: web/client/selectors/layers.js [hunk 4]
```diff
@@ -33,14 +42,7 @@ const layerSelectorWithMarkers = createSelector(
...
                     overrideOLStyle: true,
-                    style: {
-                        iconUrl: "https://cdn.rawgit.com/pointhi/leaflet-color-markers/master/img/marker-icon-red.png",
-                        shadowUrl: 'https://cdnjs.cloudflare.com/ajax/libs/leaflet/0.7.7/images/marker-shadow.png',
-                        iconSize: [25, 41],
-                        iconAnchor: [12, 41],
-                        popupAnchor: [1, -34],
-                        shadowSize: [41, 41]
-                    }
+                    style: isObject(geocoder.style) && !isEmpty(geocoder.style) && {...deafaultIconStyle, ...geocoder.style} || deafaultIconStyle
```
Comment (MV88) [inline web/client/selectors/layers.js]: typo deafaultIconStyle

File: web/client/plugins/Search.jsx [hunk 5]
```diff
@@ -66,13 +66,39 @@ const ToggleButton = require('./searchbar/ToggleButton');
  * {
  *  "name": "Search",
  *  "cfg": {
- *    "withToggle": ["max-width: 768px", "min-width: 768px"]
+ *      "withToggle": ["max-width: 768px", "min-width: 768px"],
+ *      "resultsStyle": {
+ *          "iconUrl": "https://cdn.rawgit.com/pointhi/leaflet-color-markers/master/img/marker-icon-red.png",
+ *          "shadowUrl": "https://cdnjs.cloudflare.com/ajax/libs/leaflet/0.7.7/images/marker-shadow.png",
```
Comment (MV88) [inline web/client/plugins/Search.jsx]: 077

**Pull request merged**